spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitAdmin.java

```diff
@@ -220,6 +229,9 @@ public void declareExchange(final Exchange exchange) {
 		try {
 			this.rabbitTemplate.execute(channel -> {
 				declareExchanges(channel, exchange);
+				if (this.redeclareManualDeclarations) {
+					this.manualDeclarables.put(exchange.getName(), exchange);
```

Comment:
Doesn't this look like even beans are added to the `this.manualDeclarables` ?
I mean I don't see a difference in the logic when admin declares a bean or this is a manual call of this method...

Reply:
Oops; good catch.

Reply:
Actually, its ok; the admin doesn't call these methods internally - it uses `declareQueues()` etc.

But I will add a test to ensure that doesn't change in the future.